core/src/impl/Kokkos_ViewMapping.hpp

```diff
@@ -2835,6 +2835,7 @@ struct ViewValueFunctor<ExecSpace, ValueType, false /* is_scalar */> {
   ValueType* ptr;
   size_t n;
   bool destroy;
+  std::string name;
```

Comment:
Remember that this gets copied on to the device (well, as bits anyway, not as the copy constructor, so all hell would break loose if we copied `ViewValueFunctor` on the device at any point or even tried to access the `name` member). I'm not sure how I feel about having this as a member. Do we have this in other places in Kokkos?

Reply:
@dhollman, I'll see what testing shows. To be upfront about it, I don't understand the overall structure of how this file works, this is very much a "what mechanisms get testing to pass" kind of PR for me. If you've got any other answers than this one I'd be happy to try it out.

I addressed the other comments, I think, though

Reply:
Well at the very least, there should be a comment with something like "don't use this on the device because it got bit-copied" or something like that.

Reply:
This is actually problematic for the SYCL backend because it makes `ViewValueFunctor` non trivially copyable.

Reply:
The amount of non-trivially copyable Functors in real code is virtually endless. And Views are still non-trivially copyable aren't they? I am not sure how worried I am about this. But the name of a View is anyway limited in size. We could just stick the same constant size array of char here.